test/plots/scatterPlotTests.ts

```diff
@@ -2,395 +2,399 @@
+        let allCircles = plot.selections();
+        assert.strictEqual(allCircles.size(), 4, "all circles retrieved");
+        let selectionData = allCircles.data();
+        assert.includeMembers(selectionData, data, "first dataset data in selection data");
+        assert.includeMembers(selectionData, data2, "second dataset data in selection data");
+
         svg.remove();
       });
 
-      it("rendering is idempotent", () => {
-        circlePlot.render();
-        circlePlot.render();
-        circlePlot.selections().each(getCirclePlotVerifier());
-        assert.strictEqual(circlesInArea, 10, "10 circles were drawn");
+      it("selects the closest data point", () => {
+        plot.addDataset(dataset2);
+        plot.renderTo(svg);
+        let closest = plot.entityNearest({
+          x: xScale.scale(0) + 1,
+          y: yScale.scale(0) + 1
+        });
+
+        assert.deepEqual(closest.datum, dataset.data()[0], "correct datum has been retrieved");
+
         svg.remove();
       });
 
-      describe("after the scale has changed", () => {
-        beforeEach(() => {
-          xScale.domain([0, 3]);
-          yScale.domain([0, 9]);
+      it("ignores off-plot data points when retrieving the nearest Entity", () => {
+        plot.addDataset(dataset2);
+        plot.renderTo(svg);
+        yScale.domain([0, 1.9]);
+
+        let closest = plot.entityNearest({
+          x: xScale.scale(1),
+          y: xScale.scale(2)
         });
+        assert.deepEqual(closest.datum, dataset.data()[1], "correct datum has been retrieved");
 
-        it("the circles re-rendered properly", () => {
-          let circles = circlePlot.selections();
-          circles.each(getCirclePlotVerifier());
-          assert.strictEqual(circlesInArea, 4, "four circles were found in the render area");
-          svg.remove();
+        svg.remove();
+      });
+
+      it("can retrieve Entities in a certain range", () => {
+        plot.addDataset(dataset2);
+        plot.renderTo(svg);
+
+        let entities = plot.entitiesIn({
+          min: xScale.scale(1),
+          max: xScale.scale(1)
+        }, {
+          min: yScale.scale(1),
+          max: yScale.scale(1)
+        });
+
+        assert.lengthOf(entities, 1, "only one Entity has been retrieved");
+        assert.deepEqual(entities[0].datum, { x: 1, y: 1 }, "correct datum has been retrieved");
+
+        svg.remove();
+      });
+
+      it("does not return Entities whose center lies outside the range", () => {
+        plot.addDataset(dataset2);
+        plot.renderTo(svg);
+
+        let entities = plot.entitiesIn({
+          min: xScale.scale(1.001),
+          max: xScale.scale(1.001)
+        }, {
+          min: yScale.scale(1.001),
+          max: yScale.scale(1.001)
         });
+
+        assert.lengthOf(entities, 0, "no Entities retrieved");
+
+        svg.remove();
+      });
+
+      it("can retrieve Entities in a certain bounds", () => {
+        plot.addDataset(dataset2);
```

Comment:
We have a dedicated `it()` test that does that. This test is just `"can retrieve Entities in a certain bounds"` There is no mention of multiple datasets in the title.